bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/XPathHelper.java

```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ * Copyright (c) 2025, 2025 Hannes Wellmann and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Hannes Wellmann - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.e4.ui.model;
+
+import java.util.stream.Stream;
+import org.eclipse.e4.emf.xpath.XPathContextFactory;
+import org.eclipse.e4.ui.model.application.MApplicationElement;
+
+public class XPathHelper {
```

Comment:
I think it would be fair if PDE will just call this method then

Reply:
I thought about this as well. Then lets keep it simple. I added the e4.tools bundle in PDE as a friend of the containing package.